composer.json

```diff
@@ -27,6 +27,7 @@
         "hautelook/templated-uri-bundle": "^2.1",
         "pagerfanta/pagerfanta": "^2.0",
         "ocramius/proxy-manager": "^2.1",
+        "doctrine/orm": "^2.7",
```

Comment:
Do we rely on `doctrine/orm` in every version of Ibexa DXP?

I think that instead of hard reliance on this dependency this should be moved into `required-dev` (since it's needed for testing purposes). This would unfortunately mean that the code introduced to handle this case should also be able to "switch itself off" if the dependenency is not there.

Reply:
I don't think we do optional dependencies. This introduces too much mess to the codebase.

Reply:
Then there is no point in component separation in the first place, if everything has to always be installed and loaded.

From my perspective it's as simple as not loading certain services if they are not required. Clearly the absence of doctrine/orm package would make it entirely unnecessary to go looking for EntityManagers. That's not mess, that's clear separation.

Unfortunately as I've checked, doctrine/orm is installed in our base Ibexa DXP (not sure about OSS version), so I'll yield my comment, but only due to this fact.

Reply:
AFAIK Entity Managers are part of doctrine/orm, so the dependency should be there.

Reply:
@alongosz All I'm saying is that this makes the kernel dependent on doctrine/orm whether one likes it or not. So I suggested making the services / compiler passes needed for site access aware Entity Managers to work only if the doctrine/orm package is installed from some other source.
This would result in one less package being installed if not requested, but points were raised that it would make the lib more complex and increase the maintenance. And since most projects do include doctrine/orm anyway (and it's installed with Ibexa DXP), I simply concede.